### PR TITLE
Integrating hydra with DQN

### PR DESCRIPTION
**Pull request by BoboBananas**
Integrated hydra with DQN using structured configs.

### CODE REVIEW

File: examples/ddpg/ddpg.py [hunk 1]
```diff
@@ -96,50 +84,50 @@ def main(args):
...
+        cfg=cfg,
         device=device,
     )
-    loss_module, target_net_updater = make_ddpg_loss(model, args)
+    loss_module, target_net_updater = make_ddpg_loss(model, cfg)
 
     actor_model_explore = model[0]
-    if args.ou_exploration:
-        if args.gSDE:
+    if cfg.ou_exploration:
```
Comment (vmoens) [inline examples/ddpg/ddpg.py:117]: can we delete `ou_exploration` and `gSDE` from DQN? they make no sense here. Ideally they shouldn't even be in the config

File: examples/ppo/ppo.py [hunk 2]
```diff
@@ -10,7 +10,6 @@
 import hydra
```
Comment (vmoens) [inline examples/ppo/ppo.py:10]: I'm not sure where do those changes in PPO come from in this PR

File: torchrl/trainers/helpers/models.py [hunk 3]
```diff
@@ -1351,3 +1351,5 @@ class DiscreteModelConfig:
     # whether a distributional loss should be used.
     atoms: int = 51
     # number of atoms used for the distributional loss
+    gSDE: bool = False
```
Comment (vmoens) [inline torchrl/trainers/helpers/models.py:1354]: Yeah let's not do that :) No gSDE for discrete actions

**Review by vmoens — COMMENTED** [review]
Let's see what we can do about gSDE

File: torchrl/trainers/helpers/models.py (hunk 3 above)
Comment (BoboBananas) [reply torchrl/trainers/helpers/models.py]: https://github.com/facebookresearch/rl/pull/201/commits/03b8c78b7c39cfd40d4d927d08339ce6fe70fc2c

Comment (BoboBananas) [reply torchrl/trainers/helpers/models.py]: I removed epsilon greedy wrapper which removed the error with action_value being None

File: examples/dqn/dqn.py [hunk 4]
```diff
@@ -103,17 +101,8 @@ def main(cfg: "DictConfig"):
     )
 
     loss_module, target_net_updater = make_dqn_loss(model, cfg)
-    model_explore = EGreedyWrapper(model, annealing_num_steps=cfg.annealing_frames).to(
```
Comment (vmoens) [inline examples/dqn/dqn.py]: I think we still want that, that's the typical exploration technique for dqn

File: examples/dqn/dqn.py [hunk 5]
```diff
@@ -124,7 +113,7 @@ def main(cfg: "DictConfig"):
 
     collector = make_collector_offpolicy(
         make_env=create_env_fn,
-        actor_model_explore=model_explore,
```
Comment (vmoens) [inline examples/dqn/dqn.py:129]: let's keep model_explore

**Review by vmoens — COMMENTED** [review]
EGreedyWrapper should still be used

**Pull request merged**